aexpect/ops.py

```diff
@@ -0,0 +1,479 @@
+# This Python file uses the following encoding: utf-8
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Intra2net AG and aexpect contributors
+
+"""
+A list of frequent operations performed through remote sessions.
+
+Most commands passed to a session are routine invocations of common userland
+tools. Nevertheless some of them require a certain effort to set up because
+they are squeezed through the eye of the needle, the shell session.
+
+The functionality in this module assumes a guest with the ordinary userland
+tools available. Some of it might just work on Windows guests if the proper
+environment (i. e. Cygwin) is present.
+
+This module aims to provide convenient wrappers for butter-and-bread
+functionality on a Linux system. Currently, the implemented functions fall
+under four categories: ``grep``, ``stat``, ``test``, and common file ops.
+
+grep:
+Standard case search operations. It is always assumed that binaries in the path
+should be searched (the ``-a`` flag).
+
+stat:
+Allows passing an optional format argument (``-c``). Custom wrappers for
+extracting *atime*, *ctime*, and *mtime*, etc exist.
+
+test:
+Allows for testing existence, permissions and other properties of files.
+
+file ops:
+Simple file operations executed through sessions. Most functions boil down to
+executing simple commands like ls, tar, md5sum, etc. If these operations return
+non-0, a :py:class:`RuntimeError` is raised, containing the commands's error
+message. All functions :py:func:`shlex.quote` their args for better security.
```

Comment:
It'd be nice to explain here that this module runs commands on your raw ShellSessions and although we do usually (or always?) quote the main arguments, we do not perform complex checks for escapes and that using of this module might result in data loss or worse.

Reply:
Will do, I hope if you spot any important enough case when we don't `shlex.quote` still and will be good if we do despite this disclaimer, you will let me know.

Reply:
Thanks, I don't think we should. With advanced cases people should take the responsibility themselves...